Test fails when run after other tests

Currently, test `IndexYamlBuilderTest.testChartIndexPassedCorrectly` fails when run after tests `HelmRecipeTest.haEnabledProxyRepository` and `CreateIndexServiceImplTest.testIndexYamlBuiltEvenWhenNoAssets`. I can make the test fail by running:
```
mvn test -Dtest=HelmRecipeTest#haEnabledProxyRepository,CreateIndexServiceImplTest#testIndexYamlBuiltEvenWhenNoAssets,IndexYamlBuilderTest#testChartIndexPassedCorrectly
```
in the nexus-repository-helm module. The test fails with the following exception:

```
[ERROR] testChartIndexPassedCorrectly(org.sonatype.repository.helm.internal.orient.metadata.IndexYamlBuilderTest)  Time elapsed: 0.173 s  <<< FAILURE!
org.mockito.exceptions.verification.WantedButNotInvoked:

Wanted but not invoked:
yamlParser.write(
    <any>,
    <Capturing argument>
);
-> at org.sonatype.repository.helm.internal.orient.metadata.IndexYamlBuilderTest.testChartIndexPassedCorrectly(IndexYamlBuilderTest.java:66)
Actually, there were zero interactions with this mock.

        at org.sonatype.repository.helm.internal.orient.metadata.IndexYamlBuilderTest.testChartIndexPassedCorrectly(IndexYamlBuilderTest.java:66)

[INFO]
[INFO] Results:
[INFO]
[ERROR] Failures:
[ERROR]   IndexYamlBuilderTest.testChartIndexPassedCorrectly:66
Wanted but not invoked:
yamlParser.write(
    <any>,
    <Capturing argument>
);
-> at org.sonatype.repository.helm.internal.orient.metadata.IndexYamlBuilderTest.testChartIndexPassedCorrectly(IndexYamlBuilderTest.java:66)
Actually, there were zero interactions with this mock.
```

--- a/nexus-repository-helm/src/test/java/org/sonatype/repository/helm/internal/orient/metadata/IndexYamlBuilderTest.java
+++ b/nexus-repository-helm/src/test/java/org/sonatype/repository/helm/internal/orient/metadata/IndexYamlBuilderTest.java
@@ -12,6 +12,7 @@
  */
 package org.sonatype.repository.helm.internal.orient.metadata;
 
+import java.io.ByteArrayInputStream;
 import java.io.InputStream;
 import java.io.OutputStream;
 
@@ -21,6 +22,7 @@ import org.sonatype.nexus.repository.storage.TempBlob;
 import org.sonatype.repository.helm.internal.metadata.ChartIndex;
 import org.sonatype.repository.helm.internal.util.YamlParser;
 
+import org.apache.commons.io.IOUtils;
 import org.junit.Before;
 import org.junit.Test;
 import org.mockito.ArgumentCaptor;
@@ -51,6 +53,7 @@ public class IndexYamlBuilderTest
 
   @Before
   public void setUp() throws Exception {
+    setupRepositoryMock();
     underTest = new IndexYamlBuilder(this.yamlParser);
     index = new ChartIndex();
   }
@@ -78,5 +81,14 @@ public class IndexYamlBuilderTest
 
   private void initializeStorageFacet() {
     when(storageFacet.createTempBlob(any(InputStream.class), eq(HASH_ALGORITHMS))).thenReturn(tempBlob);
+  }
+
+  private void setupRepositoryMock() {
+    when(storageFacet.createTempBlob(any(InputStream.class), any(Iterable.class))).thenAnswer(args -> {
+      InputStream inputStream = (InputStream) args.getArguments()[0];
+      byte[] bytes = IOUtils.toByteArray(inputStream);
+      when(tempBlob.get()).thenReturn(new ByteArrayInputStream(bytes));
+      return tempBlob;
+    });
   }
 }

--- a/nexus-repository-helm/src/test/java/org/sonatype/repository/helm/internal/orient/metadata/IndexYamlBuilderTest.java
+++ b/nexus-repository-helm/src/test/java/org/sonatype/repository/helm/internal/orient/metadata/IndexYamlBuilderTest.java
@@ -53,13 +53,13 @@ public class IndexYamlBuilderTest
 
   @Before
   public void setUp() throws Exception {
-    setupRepositoryMock();
     underTest = new IndexYamlBuilder(this.yamlParser);
     index = new ChartIndex();
   }
 
   @Test
   public void testChartIndexPassedCorrectly() throws Exception {
+    setupRepositoryMock();
     ArgumentCaptor<InputStream> captorStorage = ArgumentCaptor.forClass(InputStream.class);
     ArgumentCaptor<ChartIndex> captor = ArgumentCaptor.forClass(ChartIndex.class);
 

